src/hlstrafe.cpp

```diff
@@ -1958,6 +1971,53 @@ namespace HLStrafe
 			else
 				out.Pitch += vars.Frametime * changeRate;
 		}
+		if (curState.ChangeTargetYawOffsetOver > 0) {
+			float vel_yaw = 0.0;
+			if (!IsZero<float, 2>(player.Velocity))
+				vel_yaw = Atan2(player.Velocity[1], player.Velocity[0]) * M_RAD2DEG;
+			else
+				vel_yaw = yaw;
+
+			double constraints = 0;
+
+			switch (curState.Parameters.Type) {
+				case HLTAS::ConstraintsType::VELOCITY:
+					constraints = curState.Parameters.Parameters.Velocity.Constraints;
+					break;
+				case HLTAS::ConstraintsType::VELOCITY_AVG:
+					constraints = curState.Parameters.Parameters.VelocityAvg.Constraints;
+					break;
+				case HLTAS::ConstraintsType::VELOCITY_LOCK:
+					constraints = curState.Parameters.Parameters.VelocityLock.Constraints;
+					break;
+				case HLTAS::ConstraintsType::YAW:
+					constraints = curState.Parameters.Parameters.Yaw.Constraints;
+					break;
+				case HLTAS::ConstraintsType::YAW_RANGE:
+					constraints = 0.1;
+					break;
+				default:
+					assert(false);
+					break;
+			}
+
+			curState.Parameters.Type = HLTAS::ConstraintsType::YAW;
+			curState.Parameters.Parameters.Yaw.Constraints = constraints;
+
+			float targetValue = static_cast<float>(NormalizeDeg(vel_yaw + curState.ChangeTargetYawOffsetValue));
+			float difference = static_cast<float>(GetAngleDifference(yaw, targetValue));
+
+			float changeRate = difference / curState.ChangeTargetYawOffsetOver;
+			curState.ChangeTargetYawOffsetOver = std::max(0.f, curState.ChangeTargetYawOffsetOver - vars.Frametime);
+
+			if (curState.ChangeTargetYawOffsetOver == 0) {
+				curState.Parameters.Parameters.Yaw.Yaw = targetValue;
+				curState.Parameters.Type = HLTAS::ConstraintsType::VELOCITY_LOCK;
+				curState.Parameters.Parameters.VelocityLock.Constraints = 0;
+			}
+			else
+				curState.Parameters.Parameters.Yaw.Yaw = yaw + vars.Frametime * changeRate;
```

Comment:
I think instead of all of this you should add some kind of `curState.TargetYawOffset` and then just add it to the angle returned from `ComputeConstraints` / `ExactAngleConstraints`. That should be much less code and work for all target yaw types.

Reply:
Hmm, let me think about this. I just divide the work so the gradient is done by YAW strafing like how `change target_yaw` works and then it switches to VECTORIAL strafing after the gradient is finished. It is definitely simpler this way for me as I can know less and do things. But I think I can take your suggestion from the other comment to make a helper that can return the yaw gradient but still be in VECTORIAL strafing.